packages/svelte/src/internal/client/dom/elements/bindings/shared.js

```diff
@@ -54,15 +54,20 @@ export function listen_to_event_and_reset_event(element, event, handler, on_rese
 
 	if (!listening_to_form_reset) {
 		listening_to_form_reset = true;
-		document.addEventListener('reset', (evt) => {
-			requestAnimationFrame(() => {
-				if (!evt.defaultPrevented) {
-					for (const e of /**@type {HTMLFormElement} */ (evt.target).elements) {
-						// @ts-expect-error
-						e.__on_r?.();
+		document.addEventListener(
+			'reset',
+			(evt) => {
+				requestAnimationFrame(() => {
```

Comment:
Why is this using RAF, by the way?

The logic then is bound to UI updates which can potentially trigger much later if the window is not visible during the reset.

Reply:
we need to have a timeout because the dom property will not be updated to the new value when called immediately, but `Promise.resolve()` also is enough, so I switched to that.